tests/workflow_utils/test_run_cset_recipe.py

```diff
@@ -0,0 +1,217 @@
+    }
+    actual = run_cset_recipe.subprocess_env()
+    for expected_item in expected.items():
+        assert expected_item in actual.items()
+
+
+def test_recipe_file(monkeypatch, tmp_working_dir):
+    """Bundled recipe file is written to disk."""
+    monkeypatch.setenv("CSET_RECIPE_NAME", "CAPE_ratio_plot.yaml")
+    recipe_path = run_cset_recipe.recipe_file()
+    assert recipe_path == "CAPE_ratio_plot.yaml"
+    assert Path(recipe_path).is_file()
+
+
+def test_recipe_id(monkeypatch, tmp_working_dir):
+    """Extract recipe ID from recipe file."""
+
+    def mock_recipe_file():
+        with open("recipe.yaml", "wt", encoding="UTF-8") as fp:
+            fp.write("title: Recipe Title\nparallel: [{operator: misc.noop}]")
+        return "recipe.yaml"
+
+    def mock_subprocess_env():
+        return os.environ
+
+    monkeypatch.setattr(run_cset_recipe, "recipe_file", mock_recipe_file)
+    monkeypatch.setattr(run_cset_recipe, "subprocess_env", mock_subprocess_env)
+    expected = "recipe_title"
+    actual = run_cset_recipe.recipe_id()
+    assert actual == expected
+
+
+def test_recipe_id_invalid_recipe(monkeypatch, tmp_working_dir):
+    """Invalid recipe file raises exception."""
+
+    def mock_recipe_file():
+        with open("recipe.yaml", "wt", encoding="UTF-8") as fp:
+            fp.write("Not a recipe!")
+        return "recipe.yaml"
+
+    def mock_subprocess_env():
+        return os.environ
+
+    monkeypatch.setattr(run_cset_recipe, "recipe_file", mock_recipe_file)
+    monkeypatch.setattr(run_cset_recipe, "subprocess_env", mock_subprocess_env)
+    with pytest.raises(subprocess.CalledProcessError):
+        run_cset_recipe.recipe_id()
+
+
+def test_output_directory(monkeypatch):
+    """Output directory correctly interpreted."""
+
+    def mock_recipe_id():
+        return "recipe_id"
+
+    monkeypatch.setattr(run_cset_recipe, "recipe_id", mock_recipe_id)
+    monkeypatch.setenv("CYLC_WORKFLOW_SHARE_DIR", "/share")
+    actual = run_cset_recipe.output_directory()
+    expected = "/share/web/plots/recipe_id"
+    assert actual == expected
+
+
+def test_data_directory(monkeypatch):
+    """Data directory correctly interpreted."""
+    monkeypatch.setenv("CYLC_WORKFLOW_SHARE_DIR", "/share")
+    monkeypatch.setenv("CYLC_TASK_CYCLE_POINT", "20000101T0000Z")
+    expected = "/share/cycle/20000101T0000Z/data"
+    actual = run_cset_recipe.data_directory()
+    assert actual == expected
+
+
+def test_create_diagnostic_archive(tmp_path):
+    """Create ZIP archive of output."""
+    # Create dummy output files.
+    files = {"one", "two", "three"}
+    for filename in files:
+        (tmp_path / filename).touch()
+
+    run_cset_recipe.create_diagnostic_archive(tmp_path)
+    archive_path = tmp_path / "diagnostic.zip"
+    assert archive_path.is_file()
+    with zipfile.ZipFile(archive_path, "r") as archive:
+        assert set(archive.namelist()) == files
+
+
+def test_entrypoint_parallel(monkeypatch):
```

Comment:
Brief line on what this does? Not clear to me.

Reply:
This is a test I forgot to write. Good catch.

Reply:
I've added some a comment (a7e7cf7628ca5b477dd15e9d2378007dadaa02a8) and expanded some docstrings (5724650bb9cfafa90fbe8823149775cbbdb971d2). Does that clarify it?